include/micm/util/cuda_vector_matrix.hpp

```diff
@@ -151,6 +151,10 @@ namespace micm
     {
       return this->handle_;
     }
+    void SetNumberofGridCells(const std::size_t n)
+    {
+      this->vector_matrix_param_.number_of_grid_cells_ = n;
+    }
```

Comment:
I would recommend having the `number_of_grid_cells_` be set during construction, similar to how `number_of_elements_` is set. Having a separate function that needs to be called to set the number of grid cells would let instances of this class be an invalid state until this function is called, which can lead to problems.

Reply:
Thanks for the suggestion. Done.